### ISSUE TITLE
Trouble starting inventory-service

### ISSUE THREAD

**Issue by sdoxsee** [kbastani/spring-cloud-event-sourcing-example#4]
After running...

```
$ docker-machine create env event-source-demo --driver virtualbox --virtualbox-memory "11000" --virtualbox-disk-size "100000"
$ eval "$(docker-machine env event-source-demo)"
$ sh run.sh
```

I get the following exception:

```
inventory-service_1     | Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
inventory-service_1     |   at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:62)
inventory-service_1     |   at java.lang.Thread.run(Thread.java:745)
inventory-service_1     | Caused by: java.lang.reflect.InvocationTargetException
inventory-service_1     |   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
inventory-service_1     |   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
inventory-service_1     |   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
inventory-service_1     |   at java.lang.reflect.Method.invoke(Method.java:498)
inventory-service_1     |   at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:54)
inventory-service_1     |   ... 1 more
inventory-service_1     | Caused by: java.lang.IllegalStateException: Failed to execute CommandLineRunner
inventory-service_1     |   at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:809)
inventory-service_1     |   at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:790)
inventory-service_1     |   at org.springframework.boot.SpringApplication.afterRefresh(SpringApplication.java:777)
inventory-service_1     |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
inventory-service_1     |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:1191)
inventory-service_1     |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:1180)
inventory-service_1     |   at demo.InventoryApplication.main(InventoryApplication.java:29)
inventory-service_1     |   ... 6 more
inventory-service_1     | Caused by: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "meta" (class org.neo4j.ogm.session.result.RowModelResult), not marked as ignorable (one known property: "row"])
inventory-service_1     |  at [Source: {"row":[0],"meta":[null]}]}; line: 1, column: 20] (through reference chain: org.neo4j.ogm.session.result.RowModelResult["meta"])
inventory-service_1     |   at org.neo4j.ogm.session.response.RowModelResponse.next(RowModelResponse.java:48)
inventory-service_1     |   at org.neo4j.ogm.session.response.SessionResponseHandler.updateObjects(SessionResponseHandler.java:93)
inventory-service_1     |   at org.neo4j.ogm.session.delegates.SaveDelegate.save(SaveDelegate.java:69)
inventory-service_1     |   at org.neo4j.ogm.session.delegates.SaveDelegate.save(SaveDelegate.java:43)
inventory-service_1     |   at org.neo4j.ogm.session.Neo4jSession.save(Neo4jSession.java:386)
inventory-service_1     |   at org.springframework.data.neo4j.repository.GraphRepositoryImpl.save(GraphRepositoryImpl.java:53)
inventory-service_1     |   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
inventory-service_1     |   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
inventory-service_1     |   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
inventory-service_1     |   at java.lang.reflect.Method.invoke(Method.java:498)
inventory-service_1     |   at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.executeMethodOn(RepositoryFactorySupport.java:483)
inventory-service_1     |   at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.doInvoke(RepositoryFactorySupport.java:468)
inventory-service_1     |   at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.invoke(RepositoryFactorySupport.java:440)
inventory-service_1     |   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
inventory-service_1     |   at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:61)
inventory-service_1     |   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
inventory-service_1     |   at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99)
inventory-service_1     |   at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:281)
inventory-service_1     |   at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
inventory-service_1     |   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
inventory-service_1     |   at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:136)
inventory-service_1     |   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
inventory-service_1     |   at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
inventory-service_1     |   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
inventory-service_1     |   at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208)
inventory-service_1     |   at com.sun.proxy.$Proxy110.save(Unknown Source)
inventory-service_1     |   at demo.config.DatabaseInitializer.populate(DatabaseInitializer.java:92)
inventory-service_1     |   at demo.InventoryApplication.lambda$commandLineRunner$0(InventoryApplication.java:45)
inventory-service_1     |   at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:806)
inventory-service_1     |   ... 12 more
inventory-service_1     | Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "meta" (class org.neo4j.ogm.session.result.RowModelResult), not marked as ignorable (one known property: "row"])
inventory-service_1     |  at [Source: {"row":[0],"meta":[null]}]}; line: 1, column: 20] (through reference chain: org.neo4j.ogm.session.result.RowModelResult["meta"])
inventory-service_1     |   at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:51)
inventory-service_1     |   at com.fasterxml.jackson.databind.DeserializationContext.reportUnknownProperty(DeserializationContext.java:839)
inventory-service_1     |   at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1045)
inventory-service_1     |   at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1352)
inventory-service_1     |   at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1330)
inventory-service_1     |   at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:264)
inventory-service_1     |   at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
inventory-service_1     |   at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3736)
inventory-service_1     |   at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2726)
inventory-service_1     |   at org.neo4j.ogm.session.response.RowModelResponse.next(RowModelResponse.java:45)
inventory-service_1     |   ... 40 more
```

**Comment by barrycommins** [comment]
Hi @sdoxsee , this seems to be caused by the fact that the latest neo4j image (version 3.0) on Docker hub is incompatible will this example.

If you change the docker-compose.yml file to set an explicit version for that image (e.g. 2.3), it should work.

I opened pull request #3 for this issue.

**Comment by sdoxsee** [comment]
Thanks @barrycommins. Your suggestion worked great for me. I should have looked more closely at the PR before posting the issue as I thought it was just to fix a Windows issue. 

I'll leave this open until the PR is merged in case someone else runs into this problem.

Thanks again!

**Comment by barrycommins** [comment]
You're right @sdoxsee, the pull request wasn't very clear. It had two issues in the one request.
I have closed the original and opened #6

**Comment by kbastani** [comment]
Solved with #6. Thanks @barrycommins
